nautobot/core/views/mixins.py

```diff
@@ -0,0 +1,808 @@
+                        "template": "import_success.html",
+                    }
+                )
+            return super().form_valid(form)
+        else:
+            data = {}
+            if self.action in ["bulk_update", "bulk_destroy"]:
+                pk_list = self.request.POST.getlist("pk")
+                table = self.table_class(self.queryset.filter(pk__in=pk_list), orderable=False)
+                if not table.rows:
+                    messages.warning(
+                        request,
+                        f"No {self.queryset.model._meta.verbose_name_plural} were selected for {self.action}.",
+                    )
+                    return redirect(self.get_return_url(request))
+
+                data.update({"table": table})
+            data.update({"form": form})
+            return Response(data)
+
+    def form_invalid(self, form):
+        """
+        Handle invalid forms.
+        """
+        data = {}
+        request = self.request
+        if self.action in ["bulk_update", "bulk_destroy"]:
+            pk_list = self.request.POST.getlist("pk")
+            table = self.table_class(self.queryset.filter(pk__in=pk_list), orderable=False)
+            if not table.rows:
+                messages.warning(
+                    request,
+                    f"No {self.queryset.model._meta.verbose_name_plural} were selected for {self.action}.",
+                )
+                return redirect(self.get_return_url(request))
+
+            data = {
+                "table": table,
+            }
+        data.update({"form": form})
+        return Response(data)
+
+    def get_object(self):
+        """
+        Returns the object the view is displaying.
+        You may want to override this if you need to provide non-standard
+        queryset lookups.  Eg if objects are referenced using multiple
+        keyword arguments in the url conf.
+        """
+        queryset = self.queryset
+        # Perform the lookup filtering.
+        lookup_url_kwarg = self.lookup_url_kwarg or self.lookup_field
+        if lookup_url_kwarg not in self.kwargs:
+            return self.queryset.model()
+        filter_kwargs = {self.lookup_field: self.kwargs[lookup_url_kwarg]}
+        obj = get_object_or_404(queryset, **filter_kwargs)
+
+        return obj
+
+    def get_extra_context(self, request, instance=None):
+        """
+        Return any additional context data for the template.
+        request: The current request
+        instance: The object being viewed
+        """
+        return {}
+
+    def get_template_name(self):
+        # Use "<app>/<model>_<action> if available, else fall back to generic templates
+        model_opts = self.model._meta
+        app_label = model_opts.app_label
+        action = self.action
+        try:
+            if action == "retrieve":
+                template_name = f"{app_label}/{model_opts.model_name}.html"
+                select_template([template_name])
+            else:
```

Comment:
```suggestion
```

Let's remove this and do the comment deprecation of the circuit templates without action names in them.